### PR TITLE
Migrate to Go modules

### LINKED ISSUE #23: Switch go modules dependency management
Go modules is now the standard dependency for the go language. Package provider should switch the standard.

### PR DESCRIPTION
**Pull request by abhinav**
This migrates automaxprocs to Go modules, keeping the old glide.yaml in
place to provide constraints to legacy ecosystems.

As part of the migration, we significantly simplify the Makefile and add
`staticcheck` as one of the linters.

Resolves #23

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,13 +1,21 @@
...
-  directories:
-    - vendor
+
+matrix:
+  include:
+    - go: "1.12"
+    - go: "1.13"
+      env: LINT=1
+
+
```
Comment (jcorbin) [inline .travis.yml]: nit: extra blank line

Comment (abhinav) [reply .travis.yml]: fixed

File: .travis.yml [hunk 2]
```diff
@@ -1,13 +1,24 @@
...
-  directories:
-    - vendor
+
+env:
+  global:
+    - GO111MODULE=on
+
+matrix:
+  include:
+    - go: "1.12"
```
Comment (prashantv) [inline .travis.yml]: nit: might want to do `1.12.x` and `1.13.x` to get the latest patch releases

File: Makefile [hunk 3]
```diff
@@ -1,57 +1,46 @@
...
 .PHONY: test
 test:
-	@.build/test.sh
+	go test -race ./...
 
-# Disable printf-like invocation checking due to testify.assert.Error()
-VET_RULES := -printf=false
+.PHONY: cover
+cover:
+	go test -race -coverprofile=cover.out -covermode=atomic -coverpkg=./... ./...
```
Comment (prashantv) [inline Makefile]: do we need `-race` for cover? since we already use `-race` in the `test` rule above

Comment (abhinav) [reply Makefile]: You're right, we don't.

**Pull request merged**